GoogleSignIn/Sources/GIDSignIn.m

```diff
@@ -632,15 +636,18 @@ - (void)authenticateInteractivelyWithOptions:(GIDSignInInternalOptions *)options
 
 - (void)authorizationRequestWithOptions:(GIDSignInInternalOptions *)options completion:
     (void (^)(OIDAuthorizationRequest *_Nullable request, NSError *_Nullable error))completion {
-  BOOL shouldCallCompletion = YES;
+  BOOL shouldCreateAuthRequest = YES;
   NSMutableDictionary<NSString *, NSString *> *additionalParameters =
       [self additionalParametersFromOptions:options];
 #if TARGET_OS_IOS && !TARGET_OS_MACCATALYST
   if (@available(iOS 14.0, *)) {
     // Only use `_appCheck` (created via singleton `+[GIDSignIn sharedInstance]` call) if
-    // `-[GIDAppCheck prepareForAppCheckWithCompletion:]` has been called
-    if ([_appCheck isPrepared]) {
-      shouldCallCompletion = NO;
+    // `GIDAppCheck` has been successfully prepared OR if the developer has attempted to configure.
+    // If former is false and the latter true, then preparation step failed for some reason; we
+    // still want to try to pass along the app check token (it just may take longer since the
+    // pre-warm step failed).
+    if ([_appCheck isPrepared] || _configureAppCheckCalled) {
```

Comment:
I _think_ this would be a breaking change for existing users. E.g., if a developer updates to a version of the SDK that includes this then `shouldCreateAuthRequest` would be `NO` unless they set up their app for App Attest (including adding the entitlement) or they call `configureWithCompletion:`.

Reply:
If a developer upgrades to the version of GSI containing this code, but doesn't onboard to AppCheck and/or doesn't call the configure method, then:

* `shouldCreateAuthRequest` will start out as `YES` per line 639
* `[_appCheck isPrepared]` will be `NO`
* `_configureAppCheckCalled` will be `NO`
* `shouldCreateAuthRequest` will still be be `YES` because checks on 649 do not pass and `shouldCreateAuthRequest` will not be set to `NO` on line 650

Thus, this code will land on line 676 and enter the success block below:

```
if (shouldCreateAuthRequest) {
  OIDAuthorizationRequest *request = [self authorizationRequestWithOptions:options
                                                      additionalParameters:additionalParameters];
  completion(request, nil);
}
```

This will create the `OIDAuthorizationRequest` as before prior to this App Check work. Am I misunderstanding you or missing something?

Reply:
Sorry, I somehow misread this as `!([_appCheck isPrepared] || _configureAppCheckCalled)`. Thanks for clarifying, this LGTM as is.